hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -585,11 +585,10 @@ void publishPartitionRuntimeState() {
     }
 
     void sendPartitionRuntimeState(Address target) {
-        if (!node.isRunning()) {
+        if (!node.isMaster()) {
```

Comment:
This was added via https://github.com/hazelcast/hazelcast/pull/14537. But while master is shutting down, it still should be able sent partition state updates.

Reply:
Yes, partition service is reset during shutdown. Master flag is cleared before shutting down services, so, if a node is not master (anymore), that means it's being shut down already.